sphinx/search/__init__.py

```diff
@@ -252,8 +252,8 @@ def __init__(self, env: BuildEnvironment, lang: str, options: Dict, scoring: str
             self.lang: SearchLanguage = SearchEnglish(options)
         elif isinstance(lang_class, str):
             module, classname = lang_class.rsplit('.', 1)
-            lang_class = getattr(import_module(module), classname)
-            self.lang = lang_class(options)
+            lang_class: Type[SearchLanguage] = getattr(import_module(module), classname)  # type: ignore[no-redef]
+            self.lang = lang_class(options)  # type: ignore[operator]
```

Comment:
it's unfortunate these ignores are necessary. I think this whole function needs a refactor to be honest